src/spring-cloud/azext_spring_cloud/_params.py

```diff
@@ -33,6 +33,10 @@ def load_arguments(self, _):
     with self.argument_context('spring-cloud create') as c:
         c.argument('location', arg_type=get_location_type(
             self.cli_ctx), validator=validate_location)
+        c.argument('reserved_cidr_range', help='A CIDR notation IP range from which to host underlying infrastructure. Can be 1 unused /14 IP range(or as large as available), or 3 unused /16 IP ranges. Required when deploying into a Virtual Network.', validator=validate_vnet)
+        c.argument('vnet', help='The name or ID of an existing Virtual Network into which to deploy the Spring Cloud instance.', validator=validate_vnet_required_parameters)
+        c.argument('app_subnet', help='The name or ID of an existing subnet in "vnet" into which to deploy the Spring Cloud app. Required when deploying into a Virtual Network.', validator=validate_vnet_required_parameters)
```

Comment:
hi @arrownj does the global VNET in core could be leveraged here? Could service commands customize the argument name in their command?

Reply:
Let me describe our API for vnet-injected here, so that @arrownj can have a better understanding of it.

If user wants to create a vnet-injected spring cloud, he needs to specify the properties

```json
spring: {
  NetworkProfile: {
    serviceRuntimeSubnetId: <a subnet resource id we say subnet-A>,
    appSubnetId: <another subnet resource id, we say subnet-B>,
    serviceCidr: <1 or 3 cidr range which has prefix requirement>
  }
}
```

The subnet-A and subnet-B should be in the same VNet. Thus here I define the CLI interface for subent-A and subnet-B as: 
* app-subnet, service-runtime-subnet: Can give a name only if vnet specified, and then can format the resource id. Or given the resource id directly.
* vnet: If a name, format the resource id in the spring-cloud's resource group. Or a given resource id.
 This can be omitted if the 2 subnets's resource id are given.